martin-mbtiles/src/errors.rs

```diff
@@ -22,11 +20,14 @@ pub enum MbtError {
     #[error("Invalid data format for MBTile file {0}")]
     InvalidDataFormat(String),
 
+    #[error("Integrity check failed for MBTile file {0} for the following reasons: \n {1:?}")]
+    FailedIntegrityCheck(String, Vec<String>),
+
     #[error("Invalid tile data for MBTile file {0}")]
     InvalidTileData(String),
 
-    #[error("Incorrect data format for MBTile file {0}; expected {1:?} and got {2:?}")]
-    IncorrectDataFormat(String, &'static [MbtType], MbtType),
+    #[error("Expected metadata value for {key} key in MBTiles file {filepath}")]
+    MetadataValueNotFound { key: String, filepath: String },
```

Comment:
I don't think this error should be this generic - technically we should almost never rely on metadata having any specific field.  Instead, lets rename this error to something like `GlobalHashValueNotFound(filepath)` -- and only use it for that case